src/VisualStudio/Core/Def/Venus/ContainedLanguage.cs

```diff
@@ -179,5 +197,27 @@ private void OnDataBufferChanged(object sender, TextContentChangedEventArgs e)
             // when primary buffer has changed to update diagnostic positions.
             _diagnosticAnalyzerService.Reanalyze(this.Workspace, documentIds: SpecializedCollections.SingletonEnumerable(this.ContainedDocument.Id));
         }
+
+        public string GetFilePathFromBufferOrHierarchy(IVsHierarchy hierarchy, uint itemid)
+        {
+            if (SubjectBuffer.CurrentSnapshot.TextBuffer.Properties.TryGetProperty<ITextDocument>(typeof(ITextDocument), out var document) &&
+                document.FilePath is string documentFilePath)
+            {
+                return documentFilePath;
+            }
+
+            if (!ErrorHandler.Succeeded(((IVsProject)hierarchy).GetMkDocument(itemid, out var filePath)))
+            {
+                // we couldn't look up the document moniker from an hierarchy for an itemid.
+                // Since we only use this moniker as a key, we could fall back to something else, like the document name.
```

Comment:
What scenario is this fallback happening in? Is the name an absolute path or something else?